### PR TITLE
ISSUE #1346 - Remember teamspaces tree state

### PR DESCRIPTION
**Pull request by dleszcz**
**Description**

- Add logic to keep state in teamspaces store
- Remove old local storage logic

### CODE REVIEW

File: frontend/routes/components/treeList/treeList.component.tsx [hunk 1]
```diff
@@ -23,6 +23,12 @@ import { Container, Headline, Details, Title, IconContainer } from './treeList.s
 import Folder from '@material-ui/icons/Folder';
 import FolderOpen from '@material-ui/icons/FolderOpen';
 
+export const treeLevels = {
```
Comment (simplypixi) [inline frontend/routes/components/treeList/treeList.component.tsx]: the name of constant should be uppercased

File: frontend/routes/teamspaces/teamspaces.component.tsx [hunk 2]
```diff
@@ -74,10 +76,12 @@ interface IProps {
 	onRevisionsClick: () => void;
 	onDownloadClick: () => void;
 	onUploadClick: () => void;
+	setState: (componentState: any) => void;
```
Comment (simplypixi) [inline frontend/routes/teamspaces/teamspaces.component.tsx]: You should provide `IState` as `componentState` type

File: frontend/routes/teamspaces/components/projectItem/projectItem.component.tsx [hunk 3]
```diff
@@ -122,8 +123,9 @@ export class ProjectItem extends React.PureComponent<IProps, IState> {
 
 		return (
 			<TreeList
+				setActiveProject={this.props.setActiveProject}
```
Comment (simplypixi) [inline frontend/routes/teamspaces/components/projectItem/projectItem.component.tsx]: Why didn't you bind this handler to `onRootClick`?

File: frontend/routes/teamspaces/teamspaces.component.tsx [hunk 4]
```diff
@@ -390,10 +380,15 @@ export class Teamspaces extends React.PureComponent<IProps, IState> {
 					})}
 				onRemoveClick={this.createRemoveProjectHandler(props.name)}
 				active={this.isActiveProject(props.name)}
+				setActiveProject={this.setActiveProject}
```
Comment (simplypixi) [inline frontend/routes/teamspaces/teamspaces.component.tsx]: Why didn't you bind this handler to `onRootClick`?

File: frontend/modules/model/model.sagas.ts [hunk 5]
```diff
@@ -46,16 +46,19 @@ export function* fetchSettings({ teamspace, modelId }) {
 export function* updateSettings({ modelData: { teamspace, project, modelId }, settings }) {
 	try {
 		const modifiedSettings = cloneDeep(settings);
+		const activeProject = project || select(selectActiveProject);
```
Comment (simplypixi) [inline frontend/modules/model/model.sagas.ts]: Is selecting an active project from store necessary?

File: frontend/routes/teamspaces/teamspaces.component.tsx (hunk 2 above)
Comment (sanmont3drepo) [reply frontend/routes/teamspaces/teamspaces.component.tsx]: I agree with @simplypixi  if your intent is to save the whole state of the component. If you only want to save the activeTeamspace and activeProject I'd change the name of the function to be more explicit of the intent.

**Comment by carmenfan** [comment]
@sanmont3drepo can you take another look?

File: frontend/routes/teamspaces/teamspaces.component.tsx [hunk 6]
```diff
@@ -127,10 +130,11 @@ export class Teamspaces extends React.PureComponent<IProps, IState> {
...
 
 	public componentWillUnmount() {
-		if (!this.props.history.location.pathname.startsWith('/viewer')) {
-			localStorage.removeItem('lastTeamspace');
-			localStorage.removeItem('lastProject');
-		}
+		this.props.setState({
+			activeTeamspace: this.state.activeTeamspace,
+			activeProject: this.state.activeProject,
+			teamspacesItems: this.state.teamspacesItems
```
[the remark this answers is not in the document]
Comment (simplypixi) [reply frontend/routes/teamspaces/teamspaces.component.tsx]: You are absolutely right! 👍 
@dleszcz Could you fix it tomorrow morning?

**Pull request merged**